systemtest/src/test/java/io/strimzi/systemtest/StrimziUpgradeST.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.test.Namespace;
+import io.strimzi.test.StrimziExtension;
+import io.strimzi.test.TestUtils;
+import io.strimzi.test.k8s.KubeClusterException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static io.strimzi.test.StrimziExtension.REGRESSION;
+
+@ExtendWith(StrimziExtension.class)
+@Namespace(StrimziUpgradeST.NAMESPACE)
+@Tag(REGRESSION)
+public class StrimziUpgradeST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(StrimziUpgradeST.class);
+
+    public static final String NAMESPACE = "strimzi-upgrade-test";
+
+    private void copyAndModify(File root) {
```

Comment:
with this name it doesn't sound that it's applying as well, maybe `copyModifyApply` or something like that?